baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/StreamFlatMapOptional.java

```diff
@@ -128,19 +126,12 @@ private Description fix(
             return Description.NO_MATCH;
         }
 
-        return Stream.<Supplier<SuggestedFix>>of(
-                        () -> getSuggestedFix(SuggestedFix.builder(), tree, state, expressionTree, ""),
-                        () -> getQualifiedSuggestedFix(tree, state, expressionTree, receiverType.getTypeArguments()),
-                        () -> getQualifiedSuggestedFix(tree, state, expressionTree, List.of(elementType)))
-                .<SuggestedFix>mapMulti((f, consumer) -> {
-                    SuggestedFix fix = f.get();
-                    if (SuggestedFixes.compilesWithFix(fix, state)) {
-                        consumer.accept(fix);
-                    }
-                })
-                .findFirst()
-                .map(fix -> buildDescription(tree).addFix(fix).build())
-                .orElse(Description.NO_MATCH);
+        // try to elide the type arguments for mapMulti if it compiles, otherwise fallback on including qualified types
+        SuggestedFix fix = Optional.of(getSuggestedFix(SuggestedFix.builder(), tree, state, expressionTree, ""))
+                .filter(f -> SuggestedFixes.compilesWithFix(f, state))
+                .orElseGet(
+                        () -> getQualifiedSuggestedFix(tree, state, expressionTree, receiverType.getTypeArguments()));
```

Comment:
If it's safe to replace `receiverType.getTypeArguments()` with `List.of(elementType)`, we can remove the `receiver` argument to this function entirely. I'd imagine that should be safe, since we're pulling the resulting stream type of the original function (e.g. what's returned by `.map(Optional::get)`) as opposed to the intermediate/input state of the stream such a function is invoked upon.